MCprep_addon/commonmcobj_parser.py

```diff
@@ -0,0 +1,181 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Optional, Tuple, TextIO
+
+class CommonMCOBJTextureType(Enum):
+    ATLAS = "ATLAS"
+    INDIVIDUAL_TILES = "INDIVIDUAL_TILES"
+
+@dataclass
+class CommonMCOBJ:
```

Comment:
Great to have this general dataclass representation. Should we also (or instead) have a blender property group version too?

I imagine the most useful thing would be to ensure the entire spec can be saved to the the root of a world import, so it can be opened later and the user could e.g. extend the world import or re-trigger opening mineways/jmc2obj with the same settings.

We could save the content as either pre-parsed details, or maybe it would be safer in a way to save the whole header as a text block, so that the MCprep addon always attemtps to parse, but in case the user has an outdated mcprep instance compared to the spec, the full spec details are still saved in case the user updates MCprep later and then goes back to reopen this old file. If we do that, much less of a need for custom property groups.

(if this is already happening, sorry for missing it - maybe link me)

Reply:
The dataclass is used to create properties after importing the OBJ, with all values used (mostly as backward/forward compatibility in case we introduce new features that depend on more of the header)